core/src/credential.rs

```diff
@@ -20,6 +32,78 @@ pub enum CredentialType {
     PermanentResidentCard,
     BankCard,
     BankAccount,
+    UniversityDegreeCredential,
+    AlumniCredential,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, Eq)]
+pub enum PresentationType {
+    #[serde(rename = "VerifiablePresentation")]
+    Common,
+    CredentialManagerPresentation,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize, Clone)]
+#[serde(untagged)]
+pub enum CredentialSubject {
+    Single(std::collections::HashMap<String, serde_json::Value>),
+    Set(Vec<std::collections::HashMap<String, serde_json::Value>>),
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize, Clone, Validate)]
+pub struct Credential {
+    #[validate(custom(validation::credential_context_validation))]
+    #[serde(rename = "@context")]
+    pub context: DocumentContext,
+
+    #[serde(rename = "id")]
```

Comment:
Is this rename directive necessary? The field itself is already called `id`

Reply:
This is a holdover from the @id days, I think its something we can remove